public/js/main.js

```diff
@@ -218,12 +218,23 @@ bingo.controller('homeController', function($scope, $http, $location, bingosocke
   };
   $scope.editCardSet = function(cardsetid) {
     //$location.path('/cardset/edit') //Make API more "RESTful" (e.g. /<object>/<action>)
-    console.log('Go write an "edit card set" controller');
+    $location.path('/cardset/edit');
   };
 
   $scope.deleteCardSet = function(cardsetid) {
-    //$http.post(...)
-    console.log('Go write a "delete card set" controller');
+    $http.post('/api/delete/cardset', {
+        cardset_id: cardsetid
+      })
+      .success(function(data) {
+        if (data.restrict) {
+          confirm('Sorry but you are not permitted to do that');
+        } else {
+          $window.location.reload();
```

Comment:
OK, Ryan

I will do it~

From: Ryan Eggert [mailto:notifications@github.com]
Sent: Saturday, May 09, 2015 3:26 PM
To: RyanEggert/olinjs-bingo
Cc: Yuzhong Huang
Subject: Re: [olinjs-bingo] Delete cardset (#43)


In public/js/main.js<https://github.com/RyanEggert/olinjs-bingo/pull/43#discussion_r29995529>:

>    };

>

>    $scope.deleteCardSet = function(cardsetid) {

> -    //$http.post(...)

> -    console.log('Go write a "delete card set" controller');

> +    $http.post('/api/delete/cardset', {

> +        cardset_id: cardsetid

> +      })

> +      .success(function(data) {

> +        if (data.restrict) {

> +          confirm('Sorry but you are not permitted to do that');

> +        } else {

> +          $window.location.reload();

Do we have to reload? Surely we could send back a list of all cardsets and update $scope.cardsets.
We may even be able to get by with, upon confirmation of deletion, remove the deleted cardset from $scope.cardsets

[ To remove an item from an array, follow the .indexOf() and .splice() process as described here<http://stackoverflow.com/a/5767357>. ]

—
Reply to this email directly or view it on GitHub<https://github.com/RyanEggert/olinjs-bingo/pull/43/files#r29995529>.